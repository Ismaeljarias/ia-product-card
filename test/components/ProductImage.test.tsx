import * as React from 'react';
import renderer from 'react-test-renderer';
import { ProductImage, ProductCard } from '../../src/components';
import { product2 } from '../data/products';

describe('Product Image', () => {
  test('should show the component with the custom image', () => {
    const wrapper = renderer.create(
      <ProductImage className="product-name" img="https://image.com" />
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test('should show the component with the product image', () => {
    const wrapper = renderer.create(
      <ProductCard product={product2}>
        {() => <ProductImage img={product2.image} />}
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
