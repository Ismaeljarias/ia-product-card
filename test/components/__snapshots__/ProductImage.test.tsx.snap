// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product Image should show the component with the custom image 1`] = `
<img
  alt="Product"
  className="productImg product-name"
  src="https://image.com"
/>
`;

exports[`Product Image should show the component with the product image 1`] = `
<div
  className="productCard undefined"
>
  <img
    alt="Product"
    className="productImg undefined"
    src="./coffee-mug.png"
  />
</div>
`;
